### PR TITLE
chore: fix log search live tests

### PR DESCRIPTION
**Pull request by Ziinc**
Fixes log search live tests. Major issues with testing is due to async-ness with Tasks.
Refactored areas namely involves:
- conn_case macro for sandboxing
- Spawning tasks on the Logflare.TasksSupervisor to allow for Task killing on test teardown
- updating ecto_sql and phoenix_sql to newer versions, as there some changes to ecto sandboxing error handling in phoenix_ecto 4.4 [exit trapping](https://github.com/phoenixframework/phoenix_ecto/commit/1d8d28a82eef1ff496851bd2fd53661398d38b99)
  - this involves removing `:nullable` and `:null` schema field options, as is no longer supported on ecto schema level
  - also includes changing ex_oauth2_provider package to a temporary fork, as the main repo has not been updated yet for phoenix_ecto v4.4
- optimizations to prevent ecto sandbox ownership errors, namely in two places
  1. SearchLV `handle_params`, where there is an attempted preload on each params change. The added `case...do` prevents an ecto `get_by` call.
  2. SearchQueryExecutor `init` and `handle_continue` : the state initialization has been shifted up to the `init` callback, to ensure that ecto queries are complete and blocks. Tests were terminating before the ecto call could complete.

These changes is part of #1178

### CODE REVIEW

File: lib/logflare_web/live/search_live/templates/logs_search.html.leex [hunk 1]
```diff
@@ -133,15 +133,15 @@
...
-        <%= if @uri_params["tailing?"] == "true" do %>
-        <span class="btn btn-primary live-pause mr-0" phx-click="soft_play">
-          <i class="fas fa-play"></i><span class="fas-in-button hide-on-mobile">Live</span>
-        </span>
-        <% else %>
-        <span class="btn btn-primary live-pause mr-0" phx-click="hard_play">
-          <i class="fas fa-play"></i><span class="fas-in-button hide-on-mobile">Live</span>
-        </span>
-        <% end %>
+          <%= if @uri_params["tailing?"] == "true" do %>
```
Comment (Ziinc) [inline lib/logflare_web/live/search_live/templates/logs_search.html.leex:136]: this is just an indentation fix for readability

File: mix.exs [hunk 2]
```diff
@@ -83,7 +83,7 @@ defmodule Logflare.Mixfile do
 
       # Oauth2 provider
       {:phoenix_oauth2_provider, "~> 0.5.1"},
-      {:ex_oauth2_provider, github: "danschultzer/ex_oauth2_provider", override: true},
+      {:ex_oauth2_provider, github: "aristamd/ex_oauth2_provider", override: true},
```
Comment (Ziinc) [inline mix.exs:87]: fork switching

File: test/logflare_web/controllers/endpoints_controller_test.exs [hunk 3]
```diff
@@ -100,41 +100,4 @@ defmodule LogflareWeb.EndpointsControllerTest do
       assert html =~ @valid_params.query
     end
   end
-
-  # this is a successful bq response retrieved manually
-  defp gen_mock_bq_response(date) when is_binary(date) do
```
Comment (Ziinc) [inline test/logflare_web/controllers/endpoints_controller_test.exs:105]: shifted bigquery mocking logic to a the test helper file

**Review by Ziinc — COMMENTED** [review]
added comments

**Review by chasers — APPROVED** [review]

**Pull request merged**